### DISCUSSION TITLE
router.back() to navigate back to parent page

### DISCUSSION THREAD

**Discussion by MeenaFoerderer** [neuefische/questions#334] [Web] [answered]
Hello,
I am building a Next.js app where I have a list of events (talks) distributed over three different days and three different rooms. The days are pages and the rooms are dynamically rendered in a component. 

I use useRouter to navigate to a details page for each of the events/talks and also to navigate back with router.back() from there to the parent page. It works in so far that it leads back to the right day, however, it doesn't lead back to the right room. 
Does anybody have an idea how to fix this?

The details page: https://github.com/MeenaFoerderer/Capstone-project/blob/main/pages/talks/%5Bid%5D.js
The rooms component: https://github.com/MeenaFoerderer/Capstone-project/blob/main/components/TalksPerRoom/index.js
Vercel: https://capstone-project-conference-app-git-details-page-meenafoerderer.vercel.app/dayOne

**Comment by actyralf** [comment, marked as answer]
I am not sure if I understand the problem - what do you mean by "leading back to the right room"? On your Vercel-Link I see a list of days and only one room ("Great Hall").

**Reply by MeenaFoerderer** [reply]
I mean that it always leads back to the "first" room which is "Great Hall". You can switch the rooms with the Prev and Next links in the footer navigation. I will work on this problem with one of our coaches today and maybe we find a solution.
